// biteship-sdk/dist/http/client.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpClient = void 0;
const axios_1 = __importDefault(require("axios"));
class HttpClient {
    constructor(config) {
        this.config = {
            baseUrl: config.baseUrl || 'https://api.biteship.com',
            timeout: config.timeout || 30000,
        };
        this.client = axios_1.default.create({
            baseURL: this.config.baseUrl,
            timeout: this.config.timeout,
            headers: {
                'Content-Type': 'application/json',
                Authorization: config.apiKey,
            },
        });
        // Request interceptor
        this.client.interceptors.request.use((config) => config, (error) => Promise.reject(error));
        // Response interceptor
        this.client.interceptors.response.use((response) => response, (error) => {
            if (error.response) {
                // API returned an error response
                return Promise.reject({
                    message: error.response.data?.message || error.message,
                    status: error.response.status,
                    data: error.response.data,
                });
            }
            else if (error.request) {
                // Request was made but no response received
                return Promise.reject({
                    message: 'No response received from server',
                    status: 0,
                });
            }
            else {
                // Something else happened
                return Promise.reject({
                    message: error.message,
                    status: 0,
                });
            }
        });
    }
    async get(url, config) {
        const response = await this.client.get(url, config);
        return response.data;
    }
    async post(url, data, config) {
        const response = await this.client.post(url, data, config);
        return response.data;
    }
    async put(url, data, config) {
        const response = await this.client.put(url, data, config);
        return response.data;
    }
    async patch(url, data, config) {
        const response = await this.client.patch(url, data, config);
        return response.data;
    }
    async delete(url, config) {
        const response = await this.client.delete(url, config);
        return response.data;
    }
}
exports.HttpClient = HttpClient;
