// biteship-sdk/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Biteship = void 0;
const client_1 = require("./http/client");
const rates_service_1 = require("./services/rates.service");
const couriers_service_1 = require("./services/couriers.service");
const locations_service_1 = require("./services/locations.service");
const maps_service_1 = require("./services/maps.service");
const draft_orders_service_1 = require("./services/draft-orders.service");
const orders_service_1 = require("./services/orders.service");
const tracking_service_1 = require("./services/tracking.service");
/**
 * Biteship SDK - Main class for interacting with Biteship API
 */
class Biteship {
    /**
     * Create a new Biteship SDK instance
     * @param config - SDK configuration
     */
    constructor(config) {
        if (!config.apiKey) {
            throw new Error('API key is required');
        }
        this.http = new client_1.HttpClient(config);
        // Initialize services
        this.rates = new rates_service_1.RatesService(this.http);
        this.couriers = new couriers_service_1.CouriersService(this.http);
        this.locations = new locations_service_1.LocationsService(this.http);
        this.maps = new maps_service_1.MapsService(this.http);
        this.draftOrders = new draft_orders_service_1.DraftOrdersService(this.http);
        this.orders = new orders_service_1.OrdersService(this.http);
        this.tracking = new tracking_service_1.TrackingService(this.http);
    }
}
exports.Biteship = Biteship;
// Export all types
__exportStar(require("./types"), exports);
// Default export
exports.default = Biteship;
