// biteship-sdk/dist/services/draft-orders.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DraftOrdersService = void 0;
class DraftOrdersService {
    constructor(http) {
        this.http = http;
    }
    /**
     * Create a draft order
     * @param request - Draft order creation data
     * @returns Promise with draft order response
     */
    async create(request) {
        return this.http.post('/v1/draft_orders', request);
    }
    /**
     * Get a draft order by ID
     * @param id - Draft order ID
     * @returns Promise with draft order response
     */
    async get(id) {
        return this.http.get(`/v1/draft_orders/${id}`);
    }
    /**
     * Update a draft order
     * @param id - Draft order ID
     * @param request - Draft order update data
     * @returns Promise with draft order response
     */
    async update(id, request) {
        return this.http.post(`/v1/draft_orders/${id}`, request);
    }
    /**
     * Delete a draft order
     * @param id - Draft order ID
     * @returns Promise with response
     */
    async delete(id) {
        return this.http.delete(`/v1/draft_orders/${id}`);
    }
    /**
     * Confirm a draft order (convert to order)
     * @param id - Draft order ID
     * @returns Promise with order response
     */
    async confirm(id) {
        return this.http.post(`/v1/draft_orders/${id}/confirm`);
    }
    /**
     * Get rates for a draft order
     * @param id - Draft order ID
     * @returns Promise with rates response
     */
    async getRates(id) {
        return this.http.get(`/v1/draft_orders/${id}/rates`);
    }
}
exports.DraftOrdersService = DraftOrdersService;
